lib/fuel-soap.js

```diff
@@ -75,8 +75,11 @@ FuelSoap.prototype.soapRequest = function (options, callback) {
 		throw new TypeError( 'options argument is required' );
 	}
 
-	var requestOptions = this.requestOptions;
-	requestOptions.headers = this.defaultHeaders;
+	var requestOptions = _.cloneDeep(this.requestOptions);
+	requestOptions.headers = _.clone(this.defaultHeaders);
+	if (options.reqOptions) {
+		_.merge(requestOptions.headers, options.reqOptions.headers);
+	}
```

Comment:
Instead of creating all these vars and deep cloning stuff, why can't the original code be changed to something like below? I just put the vars at the top of this code block, but please define all them at the top of the function.

```js
var requestOptions = this.requestOptions;
var requestHeaders = options.reqOptions && options.reqOptions.headers || {};

requestOptions.headers = _.merge({}, this.defaultHeaders, requestHeaders)
```